### PR TITLE
Feature/verification Adding verification flow

### PR DESCRIPTION
**Pull request by asgrigoryan**
Adding verification flow.

### CODE REVIEW

File: src/main/java/com/nnaltakyan/core/auth/domain/auth/service/AuthenticationService.java [hunk 1]
```diff
@@ -35,8 +40,12 @@ public AuthenticationResponse register(final RegisterRequest registerRequest)
 				.updated(registerRequest.getUpdated())
 				.build();
 		userRepository.save(user);
-		var jwt = jwtService.generateToken(user);
-		return AuthenticationResponse.builder().token(jwt).build();
+		verificationService.createOTPAndSaveInDB(user);
+		eventPublisher.publishEvent(String.valueOf(user.getId()));
+		return RegisterResponse.builder()
+				.userId(user.getId())
+				.status(user.getStatus().getStatus())
```
Comment (nareknaltakyan1) [inline src/main/java/com/nnaltakyan/core/auth/domain/auth/service/AuthenticationService.java]: I think this part will be a reason for NPE. Please check

File: src/main/java/com/nnaltakyan/core/auth/domain/user/enums/UserStatus.java [hunk 2]
```diff
@@ -0,0 +1,13 @@
+package com.nnaltakyan.core.auth.domain.user.enums;
+
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+
+
+@AllArgsConstructor
+@Getter
+public enum UserStatus {
```
Comment (nareknaltakyan1) [inline src/main/java/com/nnaltakyan/core/auth/domain/user/enums/UserStatus.java]: please move this enum to common-lib

File: src/main/java/com/nnaltakyan/core/auth/domain/user/model/User.java [hunk 3]
```diff
@@ -38,6 +39,7 @@ public class User implements UserDetails
 	private Timestamp created;
 	private Timestamp updated;
 	private Timestamp deleted;
+	private UserStatus status;
```
Comment (nareknaltakyan1) [inline src/main/java/com/nnaltakyan/core/auth/domain/user/model/User.java:39]: for enums please always use @Enumerated(EnumType.STRING)

File: src/main/java/com/nnaltakyan/core/auth/domain/verification/events/EventPublisher.java [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.ApplicationEventPublisher;
+import org.springframework.stereotype.Component;
+@Component
+public class EventPublisher {
+    @Autowired
+    private ApplicationEventPublisher eventPublisher;
+
+    public void publishEvent(String message){
```
Comment (nareknaltakyan1) [inline src/main/java/com/nnaltakyan/core/auth/domain/verification/events/EventPublisher.java]: the publisher must always have as an argument event object and not a message, please change

File: src/main/java/com/nnaltakyan/core/auth/domain/verification/events/SendVerificationEmailEvent.java [hunk 5]
```diff
@@ -0,0 +1,14 @@
+package com.nnaltakyan.core.auth.domain.verification.events;
+
+import lombok.Getter;
+import org.springframework.context.ApplicationEvent;
+
+@Getter
+public class SendVerificationEmailEvent extends ApplicationEvent {
+    private final Long userId;
+    public SendVerificationEmailEvent(Object source, String userId) {
```
Comment (nareknaltakyan1) [inline src/main/java/com/nnaltakyan/core/auth/domain/verification/events/SendVerificationEmailEvent.java]: why "Object source"?

File: src/main/java/com/nnaltakyan/core/auth/domain/verification/listener/EmailVerificationListener.java [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+
+@Component
+@Slf4j
+@AllArgsConstructor
+public class EmailVerificationListener implements ApplicationListener<SendVerificationEmailEvent> {
+
+    private final UserRepository userRepository;
+    private final VerificationRepository verificationRepository;
+    private final SendVerificationEmailKafkaProducer kafkaProducer;
+    private final static String TOPIC = "Email verification";
```
Comment (nareknaltakyan1) [inline src/main/java/com/nnaltakyan/core/auth/domain/verification/listener/EmailVerificationListener.java]: maybe we can use an enum?

File: src/main/java/com/nnaltakyan/core/auth/domain/verification/listener/EmailVerificationListener.java [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+    @Override
+    public void onApplicationEvent(SendVerificationEmailEvent event) {
+        log.info("A kafka event for verifying the email of the user with id {} is published", event.getUserId());
+        User user = userRepository.findById(event.getUserId()).orElse(null);
```
Comment (nareknaltakyan1) [inline src/main/java/com/nnaltakyan/core/auth/domain/verification/listener/EmailVerificationListener.java]: orElse(null); please change to orElseThrow

File: src/main/java/com/nnaltakyan/core/auth/domain/verification/listener/EmailVerificationListener.java [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+        Verification verification = verificationRepository.findByUserid(event.getUserId()).orElse(null);
```
Comment (nareknaltakyan1) [inline src/main/java/com/nnaltakyan/core/auth/domain/verification/listener/EmailVerificationListener.java]: same here

File: src/main/java/com/nnaltakyan/core/auth/domain/verification/listener/EmailVerificationListener.java [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+        if (Objects.nonNull(user) && Objects.nonNull(verification)){
```
Comment (nareknaltakyan1) [inline src/main/java/com/nnaltakyan/core/auth/domain/verification/listener/EmailVerificationListener.java]: I think we need to discuss this part

File: src/main/java/com/nnaltakyan/core/auth/domain/verification/model/Verification.java [hunk 10]
```diff
@@ -0,0 +1,28 @@
...
+import lombok.Data;
+import lombok.NoArgsConstructor;
+import java.sql.Timestamp;
+
+@Data
+@Builder
+@NoArgsConstructor
+@AllArgsConstructor
+@Entity
+@Table(name = "_verification")
```
Comment (nareknaltakyan1) [inline src/main/java/com/nnaltakyan/core/auth/domain/verification/model/Verification.java]: remove "_" from name

File: src/main/java/com/nnaltakyan/core/auth/domain/verification/model/Verification.java [hunk 11]
```diff
@@ -0,0 +1,28 @@
...
+public class Verification {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    private Long userid;
+    private Long otp;
```
Comment (nareknaltakyan1) [inline src/main/java/com/nnaltakyan/core/auth/domain/verification/model/Verification.java]: let's change the name to verificationCode, it's not an otp

File: src/main/java/com/nnaltakyan/core/auth/domain/verification/repository/VerificationRepository.java [hunk 12]
```diff
@@ -0,0 +1,14 @@
...
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.stereotype.Repository;
+
+import java.util.Optional;
+
+@Repository
+public interface VerificationRepository extends JpaRepository<Verification, Long> {
+    Optional<Verification> findByUserid(Long userid);
+    Optional<Verification> findLastRecordByOtp(Long otp);
```
Comment (nareknaltakyan1) [inline src/main/java/com/nnaltakyan/core/auth/domain/verification/repository/VerificationRepository.java]: no, no, no I have to always use userId for search operation

File: src/main/java/com/nnaltakyan/core/auth/domain/verification/service/VerificationService.java [hunk 13]
```diff
@@ -0,0 +1,86 @@
...
+
+@Service
+@Slf4j
+@RequiredArgsConstructor
+public class VerificationService {
+
+    private final VerificationRepository verificationRepository;
+    private final UserRepository userRepository;
+    private final static String VERIFICATION_NOT_FOUND = "Verification record not found.";
+    private final static String USER_NOT_FOUND = "User record not found.";
```
Comment (nareknaltakyan1) [inline src/main/java/com/nnaltakyan/core/auth/domain/verification/service/VerificationService.java]: I think this part we also should move to common lib

File: src/main/java/com/nnaltakyan/core/auth/domain/verification/service/VerificationService.java [hunk 14]
```diff
@@ -0,0 +1,86 @@
...
+    public void createOTPAndSaveInDB(User user){
+        final Long otp = this.generateOTP();
+        this.saveOTP(user.getId(), otp);
+    }
+
+    public Long generateOTP() {
```
Comment (nareknaltakyan1) [inline src/main/java/com/nnaltakyan/core/auth/domain/verification/service/VerificationService.java]: for verificationCode generation I think we should create a util class, what do you think?

File: src/main/java/com/nnaltakyan/core/auth/domain/verification/service/VerificationService.java [hunk 15]
```diff
@@ -0,0 +1,86 @@
...
+        final SecureRandom random = new SecureRandom();
+        Long otp = 100000 + random.nextLong(900000);
+        log.info("Generated otp: {}", otp);
+        return otp;
+    }
+
+    @Transactional
+    public void saveOTP(Long userId, Long otp){
+        long currentTimeMillis = System.currentTimeMillis();
```
Comment (nareknaltakyan1) [inline src/main/java/com/nnaltakyan/core/auth/domain/verification/service/VerificationService.java]: why?

File: src/main/java/com/nnaltakyan/core/auth/domain/verification/service/VerificationService.java [hunk 16]
```diff
@@ -0,0 +1,86 @@
...
+                .created(currentTimestamp)
+                .build();
+        verificationRepository.save(verification);
+        log.info("Otp for user with id {} saved in the DB.", userId);
+    }
+
+    @Transactional
+    public VerificationResponse verify(VerificationRequest request) throws VerificationException {
+        log.info("Verifying the user with email: {}", request.getEmail());
+        VerificationResponse verificationResponse = VerificationResponse.builder()
```
Comment (nareknaltakyan1) [inline src/main/java/com/nnaltakyan/core/auth/domain/verification/service/VerificationService.java]: I think this method needs refactoring

File: src/main/java/com/nnaltakyan/core/auth/rest/authentication/dto/VerificationResponse.java [hunk 17]
```diff
@@ -0,0 +1,16 @@
...
+import lombok.NoArgsConstructor;
+
+@Builder
+@Data
+@NoArgsConstructor
+@AllArgsConstructor
+public class VerificationResponse {
+    Long userId;
+    String email;
+    Boolean verified;
```
Comment (nareknaltakyan1) [inline src/main/java/com/nnaltakyan/core/auth/rest/authentication/dto/VerificationResponse.java]: only userId and verified is enough I think

File: src/main/java/com/nnaltakyan/core/auth/rest/enums/EmailType.java [hunk 18]
```diff
@@ -0,0 +1,12 @@
+package com.nnaltakyan.core.auth.rest.enums;
+
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+
+@Getter
+@AllArgsConstructor
+public enum EmailType {
+    VERIFICATION("verification");
```
Comment (nareknaltakyan1) [inline src/main/java/com/nnaltakyan/core/auth/rest/enums/EmailType.java]: move to common lib

File: src/main/resources/application.yml [hunk 19]
```diff
@@ -1,6 +1,8 @@
 spring:
   application:
     name: coreapi-auth
+  kafka:
+    bootstrap-servers: localhost:9092
```
Comment (nareknaltakyan1) [inline src/main/resources/application.yml]: always use env variable so we can change it via config server

File: src/main/resources/db/migration/V02__create_verification_table.sql [hunk 20]
```diff
@@ -0,0 +1,12 @@
+CREATE TABLE "_verification" (
```
Comment (nareknaltakyan1) [inline src/main/resources/db/migration/V02__create_verification_table.sql]: remove the underline from the name

**Review by nareknaltakyan1 — CHANGES_REQUESTED** [review]

File: src/main/java/com/nnaltakyan/core/auth/domain/verification/events/EventPublisher.java [hunk 21]
```diff
@@ -0,0 +1,15 @@
+package com.nnaltakyan.core.auth.domain.verification.events;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.ApplicationEventPublisher;
+import org.springframework.stereotype.Component;
+@Component
+public class EventPublisher {
+    @Autowired
+    private ApplicationEventPublisher eventPublisher;
```
Comment (nareknaltakyan1) [inline src/main/java/com/nnaltakyan/core/auth/domain/verification/events/EventPublisher.java]: private final.... and please do not use @Autowired

File: src/main/java/com/nnaltakyan/core/auth/domain/verification/events/EventPublisher.java [hunk 22]
```diff
@@ -0,0 +1,15 @@
...
+
+    public void publishEvent(String message){
+        SendVerificationEmailEvent sendVerificationEmailEvent = new SendVerificationEmailEvent(eventPublisher, message);
```
Comment (nareknaltakyan1) [inline src/main/java/com/nnaltakyan/core/auth/domain/verification/events/EventPublisher.java]: please add log

Comment (nareknaltakyan1) [inline src/main/java/com/nnaltakyan/core/auth/domain/verification/events/EventPublisher.java]: why are you passing event publisher as an argument?

File: src/main/java/com/nnaltakyan/core/auth/domain/verification/events/kafka/SendVerificationEmailKafkaProducer.java [hunk 23]
```diff
@@ -0,0 +1,16 @@
+package com.nnaltakyan.core.auth.domain.verification.events.kafka;
+
+import com.nnaltakyan.core.auth.domain.verification.events.kafka.model.VerificationEmailSendingEvent;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.kafka.core.KafkaTemplate;
+import org.springframework.stereotype.Component;
+
+@Component
+public class SendVerificationEmailKafkaProducer {
+    @Autowired
```
Comment (nareknaltakyan1) [inline src/main/java/com/nnaltakyan/core/auth/domain/verification/events/kafka/SendVerificationEmailKafkaProducer.java]: please do not use @Autowired

File: src/main/java/com/nnaltakyan/core/auth/domain/verification/service/VerificationService.java [hunk 24]
```diff
@@ -0,0 +1,86 @@
...
+                verification.setVerified(true);
+                verificationRepository.save(verification);
+                user.setStatus(UserStatus.VERIFIED);
+                userRepository.save(user);
+                verificationResponse.setVerified(true);
+                verificationResponse.setUserId(user.getId());
+            }
+            return verificationResponse;
+        }
+        catch (VerificationException e){
```
Comment (nareknaltakyan1) [inline src/main/java/com/nnaltakyan/core/auth/domain/verification/service/VerificationService.java]: please use AdviceController

File: src/main/java/com/nnaltakyan/core/auth/rest/authentication/dto/VerificationResponse.java (hunk 17 above)
Comment (nareknaltakyan1) [reply src/main/java/com/nnaltakyan/core/auth/rest/authentication/dto/VerificationResponse.java]: only verified

File: src/main/resources/db/migration/V01__create_users_table.sql [hunk 25]
```diff
@@ -8,6 +8,7 @@ CREATE TABLE "_user" (
     created TIMESTAMP NOT NULL,
     updated TIMESTAMP NOT NULL,
     deleted TIMESTAMP,
+    status VARCHAR(255) DEFAULT 'CREATED',
```
Comment (nareknaltakyan1) [inline src/main/resources/db/migration/V01__create_users_table.sql]: please add NOT NULL

**Review by ghost — APPROVED** [review]

File: src/main/java/com/nnaltakyan/core/auth/domain/user/enums/UserStatus.java (hunk 2 above)
Comment (51494306) [reply src/main/java/com/nnaltakyan/core/auth/domain/user/enums/UserStatus.java]: - common-lib[]()

**Review by ghost — APPROVED** [review]

**Review by ghost — APPROVED** [review]

File: src/main/java/com/nnaltakyan/core/auth/domain/auth/service/AuthenticationService.java (hunk 1 above)
Comment (51494306) [reply src/main/java/com/nnaltakyan/core/auth/domain/auth/service/AuthenticationService.java]: ****

**Review by ghost — APPROVED** [review]

**Review by ghost — APPROVED** [review]

**Review by ghost — APPROVED** [review]

File: src/main/java/com/nnaltakyan/core/auth/AuthApplication.java [hunk 26]
```diff
@@ -14,8 +16,10 @@
 @EnableScheduling
 // @EnableSchedulerLock(defaultLockAtMostFor = "10m")
 @Slf4j
+@RequiredArgsConstructor
 public class AuthApplication
 {
+	private final KafkaTemplate<String, String> kafkaTemplate;
```
Comment (nareknaltakyan1) [inline src/main/java/com/nnaltakyan/core/auth/AuthApplication.java]: why?

File: src/main/java/com/nnaltakyan/core/auth/domain/verification/advice/GlobalExceptionHandler.java [hunk 27]
```diff
@@ -0,0 +1,26 @@
...
+
+@ControllerAdvice
+public class GlobalExceptionHandler {
+
+    public static final String VERIFICATION_EXCEPTION_MESSAGE = "Verification failed";
+    public static final String USER_NOT_FOUND_EXCEPTION_MESSAGE = "User not found";
+    @ExceptionHandler(VerificationException.class)
+    public ResponseEntity<String> handleException(VerificationException ex) {
+        ex.printStackTrace();
+        return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).body(VERIFICATION_EXCEPTION_MESSAGE);
```
Comment (nareknaltakyan1) [inline src/main/java/com/nnaltakyan/core/auth/domain/verification/advice/GlobalExceptionHandler.java]: INTERNAL_SERVER_ERROR?

File: src/main/java/com/nnaltakyan/core/auth/domain/verification/events/EventPublisher.java [hunk 28]
```diff
@@ -0,0 +1,17 @@
...
+import org.springframework.context.ApplicationEventPublisher;
+import org.springframework.stereotype.Component;
+@Component
+@RequiredArgsConstructor
+@Slf4j
+public class EventPublisher {
+    private final ApplicationEventPublisher eventPublisher;
+
+    public void publishEvent(SendVerificationEmailEvent emailEvent){
+        log.info("Email verification event has been thrown.");
```
Comment (nareknaltakyan1) [inline src/main/java/com/nnaltakyan/core/auth/domain/verification/events/EventPublisher.java]: please add email in the log

File: src/main/java/com/nnaltakyan/core/auth/domain/verification/events/SendVerificationEmailEvent.java [hunk 29]
```diff
@@ -0,0 +1,15 @@
...
+
+import com.nnaltakyan.core.auth.domain.auth.service.AuthenticationService;
+import lombok.Getter;
+import org.springframework.context.ApplicationEvent;
+
+@Getter
+public class SendVerificationEmailEvent extends ApplicationEvent {
+
+    private final Long userId;
+    public SendVerificationEmailEvent(AuthenticationService source, Long userId) {
```
Comment (nareknaltakyan1) [inline src/main/java/com/nnaltakyan/core/auth/domain/verification/events/SendVerificationEmailEvent.java]: I don't think that there is a compulsion to use AuthenticationService

Comment (asgrigoryan) [reply src/main/java/com/nnaltakyan/core/auth/domain/verification/events/SendVerificationEmailEvent.java]: @nareknaltakyan1 we need to define the event source which is EventPublisher class in our case. I made according changes. Please take a look.

File: src/main/java/com/nnaltakyan/core/auth/domain/verification/events/EventPublisher.java (hunk 28 above)
Comment (asgrigoryan) [reply src/main/java/com/nnaltakyan/core/auth/domain/verification/events/EventPublisher.java]: @nareknaltakyan1 At the point of publishing the event I only have "userId" property in email event.

File: src/main/java/com/nnaltakyan/core/auth/domain/verification/advice/GlobalExceptionHandler.java (hunk 27 above)
Comment (asgrigoryan) [reply src/main/java/com/nnaltakyan/core/auth/domain/verification/advice/GlobalExceptionHandler.java]: @nareknaltakyan1 I changed the status code to unauthorized as this is the case when authentication fails.

File: src/main/java/com/nnaltakyan/core/auth/AuthApplication.java (hunk 26 above)
Comment (asgrigoryan) [reply src/main/java/com/nnaltakyan/core/auth/AuthApplication.java]: Most probably randomly) removed.

File: src/main/java/com/nnaltakyan/core/auth/domain/user/enums/UserStatus.java (hunk 2 above)
Comment (asgrigoryan) [reply src/main/java/com/nnaltakyan/core/auth/domain/user/enums/UserStatus.java]: Done.

File: src/main/java/com/nnaltakyan/core/auth/domain/verification/listener/EmailVerificationListener.java (hunk 6 above)
Comment (asgrigoryan) [reply src/main/java/com/nnaltakyan/core/auth/domain/verification/listener/EmailVerificationListener.java]: Done.

File: src/main/java/com/nnaltakyan/core/auth/rest/enums/EmailType.java (hunk 18 above)
Comment (asgrigoryan) [reply src/main/java/com/nnaltakyan/core/auth/rest/enums/EmailType.java]: Done.

File: src/main/java/com/nnaltakyan/core/auth/domain/verification/service/VerificationService.java (hunk 13 above)
Comment (asgrigoryan) [reply src/main/java/com/nnaltakyan/core/auth/domain/verification/service/VerificationService.java]: Done.

File: src/main/java/com/nnaltakyan/core/auth/rest/enums/EmailType.java (hunk 18 above)
Comment (nareknaltakyan1) [reply src/main/java/com/nnaltakyan/core/auth/rest/enums/EmailType.java]: if you did, please delete it

File: src/main/java/com/nnaltakyan/core/auth/domain/auth/service/AuthenticationService.java [hunk 30]
```diff
@@ -23,15 +30,31 @@ public class AuthenticationService
...
+	public RegisterResponse register(final RegisterRequest registerRequest)
 	{
 		var user = User.builder().firstName(registerRequest.getFirstName()).lastName(registerRequest.getLastName()).email(registerRequest.getEmail())
 			.password(passwordEncoder.encode(registerRequest.getPassword())).role(Role.USER).created(registerRequest.getCreated())
 			.updated(registerRequest.getUpdated()).build();
 		userRepository.save(user);
-		var jwt = jwtService.generateToken(user);
-		return AuthenticationResponse.builder().token(jwt).build();
+		verificationService.createVerificationCodeAndSaveInDB(user);
+		if (Objects.nonNull(user.getId())){
```
Comment (nareknaltakyan1) [inline src/main/java/com/nnaltakyan/core/auth/domain/auth/service/AuthenticationService.java]: I think there is no need to write if

File: src/main/java/com/nnaltakyan/core/auth/domain/auth/service/AuthenticationService.java [hunk 31]
```diff
@@ -23,15 +30,31 @@ public class AuthenticationService
...
+			SendVerificationEmailEvent sendVerificationEmailEvent = new SendVerificationEmailEvent(eventPublisher, user.getId());
```
Comment (nareknaltakyan1) [inline src/main/java/com/nnaltakyan/core/auth/domain/auth/service/AuthenticationService.java]: todo check @nareknaltakyan1

File: src/main/java/com/nnaltakyan/core/auth/domain/auth/service/AuthenticationService.java [hunk 32]
```diff
@@ -23,15 +30,31 @@ public class AuthenticationService
...
+			eventPublisher.publishEvent(sendVerificationEmailEvent);
+			if (Objects.nonNull(user.getStatus())){
+				return RegisterResponse.builder()
+						.userId(user.getId())
+						.status(user.getStatus().getStatus())
+						.email(user.getEmail()).build();
+			}
+		}
+		return RegisterResponse.builder()
```
Comment (nareknaltakyan1) [inline src/main/java/com/nnaltakyan/core/auth/domain/auth/service/AuthenticationService.java]: please remove

File: src/main/java/com/nnaltakyan/core/auth/domain/auth/service/AuthenticationService.java [hunk 33]
```diff
@@ -23,15 +30,31 @@ public class AuthenticationService
...
 			.password(passwordEncoder.encode(registerRequest.getPassword())).role(Role.USER).created(registerRequest.getCreated())
 			.updated(registerRequest.getUpdated()).build();
 		userRepository.save(user);
-		var jwt = jwtService.generateToken(user);
-		return AuthenticationResponse.builder().token(jwt).build();
+		verificationService.createVerificationCodeAndSaveInDB(user);
+		if (Objects.nonNull(user.getId())){
+			SendVerificationEmailEvent sendVerificationEmailEvent = new SendVerificationEmailEvent(eventPublisher, user.getId());
+			eventPublisher.publishEvent(sendVerificationEmailEvent);
+			if (Objects.nonNull(user.getStatus())){
```
Comment (nareknaltakyan1) [inline src/main/java/com/nnaltakyan/core/auth/domain/auth/service/AuthenticationService.java]: please remove unnecessary check

File: src/main/java/com/nnaltakyan/core/auth/domain/user/enums/UserStatus.java (hunk 2 above)
Comment (nareknaltakyan1) [reply src/main/java/com/nnaltakyan/core/auth/domain/user/enums/UserStatus.java]: please remove

File: src/main/java/com/nnaltakyan/core/auth/domain/user/exceptions/UserNotFoundException.java [hunk 34]
```diff
@@ -0,0 +1,10 @@
+package com.nnaltakyan.core.auth.domain.user.exceptions;
+
+import lombok.AllArgsConstructor;
+import lombok.Data;
+
+@AllArgsConstructor
+@Data
+public class UserNotFoundException extends RuntimeException{
```
Comment (nareknaltakyan1) [inline src/main/java/com/nnaltakyan/core/auth/domain/user/exceptions/UserNotFoundException.java]: please move to common-lib

File: src/main/java/com/nnaltakyan/core/auth/domain/verification/events/EventPublisher.java (hunk 28 above)
Comment (nareknaltakyan1) [reply src/main/java/com/nnaltakyan/core/auth/domain/verification/events/EventPublisher.java]: ok, just log userID

File: src/main/java/com/nnaltakyan/core/auth/domain/auth/service/AuthenticationService.java [hunk 35]
```diff
@@ -23,15 +27,19 @@ public class AuthenticationService
...
+	public RegisterResponse register(final RegisterRequest registerRequest)
 	{
 		var user = User.builder().firstName(registerRequest.getFirstName()).lastName(registerRequest.getLastName()).email(registerRequest.getEmail())
 			.password(passwordEncoder.encode(registerRequest.getPassword())).role(Role.USER).created(registerRequest.getCreated())
 			.updated(registerRequest.getUpdated()).build();
 		userRepository.save(user);
-		var jwt = jwtService.generateToken(user);
-		return AuthenticationResponse.builder().token(jwt).build();
+		verificationService.createVerificationCodeAndSaveInDB(user);
+		SendVerificationEmailEvent sendVerificationEmailEvent = new SendVerificationEmailEvent(eventPublisher, user.getId());
```
Comment (nareknaltakyan1) [inline src/main/java/com/nnaltakyan/core/auth/domain/auth/service/AuthenticationService.java]: todo: check @nareknaltakyan1

**Review by nareknaltakyan1 — APPROVED** [review]

**Pull request merged**